lib/remote.provider.js

```diff
@@ -1,7 +1,113 @@
 'use strict';
 
+const jstp = require('@metarhia/jstp');
+
 const { StorageProvider } = require('./provider');
+const { RemoteCursor } = require('./remote.cursor');
+
+const APP_NAME = 'GLOBAL_STORAGE';
+const INTERFACES = ['provider', 'cursor'];
+
+class RemoteProvider extends StorageProvider {
+  // Create RemoteProvider
+  // cursor - RemoteCursor instance
+  // apiPromise - jstp promise
+  constructor() {
+    super();
+    this.api = null;
+    this.connection = null;
+    this.cursors = [];
+  }
+
+  // Open StorageProvider
+  //   callback - function(err, RemoteProvider)
+  // eslint-disable-next-line no-unused-vars
+  open(options, callback) {
```

Comment:
To make the `RemoteProvider` properly configurable, `options` must be an object here, with at least two fields defined:
The first is `transport` that will be used when calling `connect()` method:
```js
jstp[options.transport].connect(/*args here*/);
```
The second is `connectionArgs` that must be an array, and that should be used to pass arguments to the `connect()` method as I've mentioned in the comment above.